torch/__init__.py

```diff
@@ -667,6 +667,11 @@ def set_default_tensor_type(t):
 
     The default floating point tensor type is initially ``torch.FloatTensor``.
 
+    .. warning::
+
+        This function is deprecated as of PyTorch 2.1, please use :func:`torch.set_default_dtype()` and
+        :func:`torch.set_default_device()` as alternatives.
+
```

Comment:
For visibility, could you phrase this at the top level in line 664? 

Right now, the warning isn't too obvious.
<img width="921" alt="image" src="https://github.com/pytorch/pytorch/assets/31798555/3f943930-8273-4eb9-80b6-6df577629c6b">
<img width="885" alt="image" src="https://github.com/pytorch/pytorch/assets/31798555/84d6acf9-9868-48f3-bd0e-b78d6dfe4975">